state/accountsEphemeralProvider.go

```diff
@@ -73,7 +74,11 @@ func (provider *AccountsEphemeralProvider) GetUserAccount(address []byte) (UserA
 	}
 
 	account, err := provider.getExistingAccountTypedAsUserAccount(address)
-	if err != nil && err != ErrAccNotFound {
+
+	var errAccountNotFoundAtBlock *ErrAccountNotFoundAtBlock
+	ok = errors.As(err, &errAccountNotFoundAtBlock)
+
+	if err != nil && !errors.Is(err, ErrAccNotFound) && !ok {
```

Comment:
I think we can make this way cleaner by having a single bool variable like:
`isExpectedError := !errors.Is(err, ErrAccNotFound) || errors.As(err, &errAccountNotFoundAtBlock)`

and below check smth like:

`if err!= nil && !isExpectedError`